lib/model/query/forms.js

```diff
@@ -290,7 +292,20 @@ createVersion.audit = (newDef, partial, form, publish) => (log) => ((publish ===
   ? log('form.update.publish', form, { oldDefId: form.currentDefId, newDefId: newDef.id })
   : log('form.update.draft.set', form, { oldDraftDefId: form.draftDefId, newDraftDefId: newDef.id }));
 createVersion.audit.withResult = true;
+createVersion.audit.logEvenIfAnonymous = true;
+
+// This is used in the rare case where we want to change and update a FormDef in place without
+// creating a new def. This is basically a wrapper around _updateDef that also logs an event.
+const replaceDef = (partial, form) => async ({ Forms }) => {
+  const { version, hash, sha, sha256 } = partial.def;
+  await Forms._updateDef(form.def, { xml: partial.xml, version, hash, sha, sha256 });
+  // all this does is changed updatedAt
+  await Forms._update(form, { updatedAt: (new Date()).toISOString() });
+};
 
+replaceDef.audit = (_, form) => (log) =>
+  log('form.update.draft.replace', form, { upgrade: 'Updated entities-version in form to 2024.1' });
```

Comment:
Add this to the list of audits that frontend knows about (non-verbose and form list)